index.bs

```diff
@@ -11266,54 +11266,101 @@ task queue=] of its associated {{BaseAudioContext}}.
 				[=Computing a block of audio|compute a block of audio=], and
 				[=Making a buffer available for reading|make it available for reading=].
 
-			4. If this {{AudioNode}} is an {{AudioWorkletNode}}, execute these
-				substeps:
+			4. If this {{AudioNode}} is an {{AudioWorkletNode}}, execute these substeps:
 
-				1. Let <var>processor</var> be the associated {{AudioWorkletProcessor}}
+				1. Let |processor| be the associated {{AudioWorkletProcessor}}
 					instance of {{AudioWorkletNode}}.
 
-				1. If {{[[callable process]]}} of <var>processor</var> is `true`,
-					execute the following steps:
-
-					1. Let <var>processFunction</var> be the result of
-						<code><a href="http://www.ecma-international.org/ecma-262/6.0/#sec-get-o-p">
-						Get</a>(O=<i><var>processor</var></i>, P="process")</code>.
-
-					1. Set {{[[callable process]]}} to be the return value of
-						<code><a href="http://www.ecma-international.org/ecma-262/6.0/#sec-iscallable">
-						IsCallable</a>(argument=<i><var>processFunction</var></i>)</code>.
-
-					1. If {{[[callable process]]}} is `true`,
-						invoke <var>processFunction</var> to
-						[=Computing a block of audio|compute a block of audio=] with the
-						argument of [=input buffer=], output buffer and
-						[=input AudioParam buffer=]. A buffer containing copies of the
-            					elements of the {{Float32Array}}s passed via the
-            					{{AudioWorkletProcessCallback/outputs!!argument}} parameter to <var>processFunction</var> is
-						<a href="#available-for-reading">made available for reading</a>.
-
-					1. At the conclusion of <var>processFunction</var>,
-						<a href="https://tc39.github.io/ecma262/#sec-toboolean"><code>ToBoolean</code></a>
-						is applied to the return value and the result is
-						assigned to the associated {{AudioWorkletProcessor}}'s
-						<a>active source</a> flag. This in turn affects whether
-						subsequent invocations of [=process()=] occur, and has
-						an impact on the lifetime of the node.
-
-					1. Else if {{[[callable process]]}} is `false`,
-						<a>queue a task</a> to the <a>control thread</a>
-						<a spec="dom" lt="fire an event">fire</a> an
-						<a href="https://www.w3.org/TR/html50/webappapis.html#the-errorevent-interface">ErrorEvent</a>
-						named <code>processorerror</code> at the associated
-						{{AudioWorkletNode}}.
-
-				1. If {{[[callable process]]}} of <var>processor</var> is `false`,
-					execute the following steps:
-
-					1. [=Making a buffer available for reading|Make a silent output buffer available for reading=].
-
-				1. Any {{Promise}} resolved within the execution of process method will
-					be queued into the microtask queue in the {{AudioWorkletGlobalScope}}.
+				1. Let |O| be the ECMAScript object corresponding to |processor|.
+
+				1. Let |processCallback| be an uninitialized variable.
+
+				1. Let |completion| be an uninitialized variable.
```

Comment:
Might as well combine processCallback and completion into one step.  Doesn't need to be two, right?